components/brave_rewards/resources/extension/brave_rewards/content_scripts/reddit.ts

```diff
@@ -0,0 +1,417 @@
+  braveTipIcon.style.display = 'inline-block'
+  braveTipIcon.style.fontSize = '18px'
+  braveTipIcon.style.fontStyle = 'normal'
+  braveTipIcon.style.height = '16px'
+  braveTipIcon.style.position = 'relative'
+  braveTipIcon.style.verticalAlign = 'baseline'
+  braveTipIcon.style.width = '16px'
+  return braveTipIcon
+}
+
+const createRedditTipActionCount = () => {
+  const braveTipActionCount = document.createElement('span')
+  braveTipActionCount.className = 'reddit-actionCount'
+  braveTipActionCount.style.color = 'inherit'
+  braveTipActionCount.style.display = 'inline-block'
+  braveTipActionCount.style.fontSize = '12px'
+  braveTipActionCount.style.fontWeight = 'bold'
+  braveTipActionCount.style.lineHeight = '1'
+  braveTipActionCount.style.marginLeft = '3px'
+  braveTipActionCount.style.position = 'relative'
+  braveTipActionCount.style.verticalAlign = 'text-bottom'
+  return braveTipActionCount
+}
+
+const createRedditTipActionCountPresentation = () => {
+  const braveTipActionCountPresentation = document.createElement('span')
+  braveTipActionCountPresentation.className = 'ProfileReddit-actionCountForPresentation'
+  braveTipActionCountPresentation.textContent = getMessage('redditTipsIconLabel')
+  return braveTipActionCountPresentation
+}
+
+const createHoverStyleElement = (isPost: boolean) => {
+  // Create style element for hover
+  const style = document.createElement('style')
+  style.innerHTML = isPost ? ':host(:hover) { background-color: rgba(215, 218, 220, 0.1) }' : '.reddit-actionButton :hover { text-decoration: underline }'
+  return style
+}
+
+const getMoreActionCommentElement = (commentElement: Element) => {
+  return commentElement.querySelector('button[aria-label="more options"]')
+}
+
+const getMoreActionPostElement = (postElement: Element) => {
+  const element = postElement.querySelector('button[aria-label="more options"]')
+  if (element) {
+    return !element.nextElementSibling && !element.previousElementSibling && element.parentElement ? element.parentElement : element
+  }
+  return null
+}
+
+const getSaveElement = (commentElement: Element) => {
+  return document.evaluate(".//button[text()='Save']", commentElement, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue as HTMLElement
+}
+
+const getPromotedSaveElement = (element: Element) => {
+  const saveElement = document.evaluate(".//span[text()='save']", element, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue as HTMLElement
+  if (saveElement && saveElement.parentElement) {
+    return saveElement.parentElement
+  }
+  return null
+}
+
+const isUsersOwnPost = (commentElement: Element) => {
+  const editElement: Node = document.evaluate(".//button[text()='edit']", commentElement, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue
+  return editElement
+}
+
+const createBraveTipActionForLegacy = (redditPost: Element) => {
```

Comment:
Would be good to standardize on "old" or "legacy", just for consistency.

Reply:
Done